IbisHardwareIGSIO/ibishardwareIGSIO.cpp

```diff
@@ -484,28 +513,38 @@ int IbisHardwareIGSIO::FindToolByName( QString name )
 
 void IbisHardwareIGSIO::AssignDeviceImageToTool( igtlioDevicePointer device, Tool * tool )
 {
-  vtkImageData * imageContent = nullptr;
-  if( IsDeviceImage( device ) )
-  {
-      igtlioImageDevicePointer imageDev = igtlioImageDevice::SafeDownCast( device );
-      imageContent = imageDev->GetContent().image;
-  }
-  else if( IsDeviceVideo( device ) )
-  {
-      igtlioVideoDevicePointer videoDev = igtlioVideoDevice::SafeDownCast( device );
-      imageContent = videoDev->GetContent().image;
-  }
-
-  if( tool->sceneObject->IsA( "CameraObject" ) )
-  {
-      vtkSmartPointer<CameraObject> cam = CameraObject::SafeDownCast( tool->sceneObject );
-      cam->SetVideoInputData( imageContent );
-  }
-  else if( tool->sceneObject->IsA( "UsProbeObject" ) )
-  {
-      vtkSmartPointer<UsProbeObject> probe = UsProbeObject::SafeDownCast( tool->sceneObject );
-      probe->SetVideoInputData( imageContent );
-  }
+    vtkImageData * imageContent = nullptr;
+
+    if( IsDeviceImage( device ) )
+    {
+        igtlioImageDevicePointer imageDev = igtlioImageDevice::SafeDownCast( device );
+        imageContent = imageDev->GetContent().image;
+    }
+    else if( IsDeviceVideo( device ) )
+    {
+        igtlioVideoDevicePointer videoDev = igtlioVideoDevice::SafeDownCast( device );
+        imageContent = videoDev->GetContent().image;
+        if(tool->flipYAxis){
+            //TODO: This shouldn't need to be done every frame. Check pipeline logic.
+            tool->flipYFilter->SetInputData( imageContent );
+            tool->flipYFilter->Update();
+        }
+    }
+
+    if( tool->sceneObject->IsA( "CameraObject" ) )
+    {
+        vtkSmartPointer<CameraObject> cam = CameraObject::SafeDownCast( tool->sceneObject );
+        if(tool->flipYAxis){
+            cam->SetVideoInputData( tool->flipYFilter->GetOutput() );
+        }else{
+            cam->SetVideoInputData( imageContent );
+        }
+    }
+    else if( tool->sceneObject->IsA( "UsProbeObject" ) )
```

Comment:
UsProbe can't have its y flipped?

Reply:
It definitely can. I'll add to the probe as well.